task/build-maven-zip/0.1/build-maven-zip.yaml

```diff
@@ -0,0 +1,194 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: build-maven-zip
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "maven-build, konflux"
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "maven-zip"
+spec:
+  description: |-
+    Build-maven-zip task builds prefetched maven artifacts into a OCI-artifact with zip bundle  and pushes the OCI-artifact into container registry.
+    In addition it will use the SBOM file in prefetch-task, pushes the SBOM file to same registry of zip oci-artifact using cosign tool.
+    Note that this task needs the output of prefetch-dependencies task. If it is not activated, there will not be any output from this task.
+  params:
+  - description: Reference of the OCI-Artifact this build task will produce.
+    name: IMAGE
+    type: string
+  - default: "maven-repository"
+    description: The root directory of the artifacts under the prefetched directory. Will be kept in the maven zip as the top directory for all artifacts.
+    name: PREFETCH_ROOT
+    type: string
+  - default: "maven-repository"
+    description: The zip bundle file name of archived artifacts
+    name: FILE_NAME
+    type: string
+  - default: ""
+    description: Delete image tag after specified time. Empty means to keep the image tag. Time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
+    name: IMAGE_EXPIRES_AFTER
+    type: string
+  - name: caTrustConfigMapName
+    type: string
+    description: The name of the ConfigMap to read CA bundle data from.
+    default: trusted-ca
+  - name: caTrustConfigMapKey
+    type: string
+    description: The name of the key in the ConfigMap that contains the CA bundle data.
+    default: ca-bundle.crt
+
+  results:
+  - description: Digest of the OCI-Artifact just built
+    name: IMAGE_DIGEST
+  - description: OCI-Artifact repository and tag where the built OCI-Artifact was pushed
+    name: IMAGE_URL
+  - description: OCI-Artifact reference of the built OCI-Artifact
+    name: IMAGE_REF
+  - name: SBOM_BLOB_URL
+    description: Reference of SBOM blob digest to enable digest-based verification from provenance
+    type: string
+  stepTemplate:
+    env:
+    - name: IMAGE
+      value: $(params.IMAGE)
+    - name: PKG_ROOT
+      value: $(params.PREFETCH_ROOT)
+    - name: FILE_NAME
+      value: $(params.FILE_NAME)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.IMAGE_EXPIRES_AFTER)
+    volumeMounts:
+      - mountPath: /shared
+        name: shared
+
+  steps:
+  - image: quay.io/konflux-ci/appstudio-utils@sha256:d0fecb67dd98d874704f2e3d20686363de7cbb42c15ad2b1d8e8c38118c50078
+    name: prepare
+    computeResources:
+      limits:
+        memory: 8Gi
+        cpu: '4'
+      requests:
+        memory: 2Gi
+        cpu: '1'
```

Comment:
Does it need that much memory? It's just creating one zip file

Reply:
I heard from user that the prefetched artifacts may contain more than 10k+ files, and the whole zip file size will be 1GB+ or even great than 2GB. So maybe 2Gi is better?

Reply:
ok